### PR TITLE
Fix errors messages in uyuniadm uninstall 

### PR DESCRIPTION
**Pull request by NamelessOne91**
Would help debugging issue #89

### CODE REVIEW

File: shared/utils/connection.go [hunk 1]
```diff
@@ -31,16 +31,14 @@ func NewConnection(backend string) *Connection {
...
 	var err error
 	if c.command == "" {
-		command := ""
-
 		switch c.backend {
 		case "podman":
 			fallthrough
 		case "podman-remote":
-			fallthrough
+			c.command = c.backend
```
Comment (cbosdo) [inline shared/utils/connection.go:38]: We want these two cases to check for the command to be in the `PATH`, so this change doesn't look OK

File: shared/utils/connection.go [hunk 2]
```diff
@@ -31,16 +31,14 @@ func NewConnection(backend string) *Connection {
...
 		case "kubectl":
-			if _, err = exec.LookPath(command); err != nil {
-				err = fmt.Errorf("backend command not found in PATH: %s", command)
+			if _, err = exec.LookPath(c.backend); err != nil {
+				err = fmt.Errorf("backend command not found in PATH: %s", c.backend)
```
Comment (cbosdo) [inline shared/utils/connection.go:41]: Here `c.backend` is usually not set: this code ensures we have the `podman` , `podman-remote` or `kubectl` command in the `PATH` before using it. So better remove this change

File: uyuniadm/cmd/uninstall/podman.go [hunk 3]
```diff
@@ -21,9 +21,9 @@ func uninstallForPodman(dryRun bool, purge bool) {
 	} else {
 		if dryRun {
 			log.Info().Msgf("Would run systemctl disable --now uyuni-server")
-			log.Debug().Msgf("Woud remove %s", podman.ServicePath)
+			log.Debug().Msgf("Would remove %s", podman.ServicePath)
```
Comment (cbosdo) [inline uyuniadm/cmd/uninstall/podman.go:24]: Thanks for spotting those typos!

File: uyuniadm/cmd/uninstall/podman.go [hunk 4]
```diff
@@ -32,7 +32,9 @@ func uninstallForPodman(dryRun bool, purge bool) {
 
 			// Remove the service unit
 			log.Debug().Msgf("Remove %s", podman.ServicePath)
-			os.Remove(podman.ServicePath)
+			if err := os.Remove(podman.ServicePath); err != nil {
+				log.Error().Err(err).Msg("Failed to remove uyuni-server.service")
+			}
```
Comment (cbosdo) [inline uyuniadm/cmd/uninstall/podman.go:37]: Thanks for adding that missing error check: this may help understanding the issue, but surely doesn't fix it.

**Review by cbosdo — CHANGES_REQUESTED** [review]
Please drop the changes in the connection code as those would be leading to a lot of other issues. I am curious to know the reasons for those changes.

**Comment by NamelessOne91** [comment]
Hi Cedric, thanks for the review.

I've dropped the changes to the connection-related code and kept only the additional error logs ATM.

As you stated in the issue's comments, the problem Michele had is likely due to not having a container named _uyuni-server_ running. This causes no backend to be picked up even if podman is present .
Is that working as intended ?

The lack of the `Msg` method call on line 28 of  [uninstall.go](https://github.com/uyuni-project/uyuni-tools/blob/main/uyuniadm/cmd/uninstall/uninstall.go)  may contribute to make this harder to spot.

Possible changes to the connection code are probably out of scope and may involve some refactoring, it might be better to discuss them as a separate topic.

**Comment by cbosdo** [comment]
> As you stated in the issue's comments, the problem Michele had is likely due to not having a container named _uyuni-server_ running. This causes no backend to be picked up even if podman is present . Is that working as intended ?

No, this is not intended, but I haven't though very deeply how we should be handling that case. We probably need to try looking for other specific files and resources deployed early, like the systemd service file, the traefik config or similar.

> The lack of the `Msg` method call on line 28 of [uninstall.go](https://github.com/uyuni-project/uyuni-tools/blob/main/uyuniadm/cmd/uninstall/uninstall.go) may contribute to make this harder to spot.
> 
> Possible changes to the connection code are probably out of scope and may involve some refactoring, it might be better to discuss them as a separate topic.

Changing those is not easy and we should probably have a smarter detection only for uninstall in case nothing worked

**Review by cbosdo — APPROVED** [review]

**Comment by cbosdo** [comment]
Could you please squash the two commits into one before I merge?

**Comment by NamelessOne91** [comment]
Done, thanks.

**Comment by cbosdo** [comment]
I'll postpone the merge until Uyuni 2023.10 is out

**Pull request merged**